### PR TITLE
Update document.js

### PR DESCRIPTION
**Pull request by randy5235**
Fix for Issue #12157

**Summary**
Allow document.set to take an object of key/value pairs for path value, per the documentation.

**Examples**

Please see issue #12157 for example code.

### CODE REVIEW

File: lib/document.js [hunk 1]
```diff
@@ -993,6 +993,12 @@ Document.prototype.overwrite = function overwrite(obj) {
  */
 
 Document.prototype.$set = function $set(path, val, type, options) {
+  if (utils.isPOJO(path) {
```
Comment (hasezoey) [inline lib/document.js:996]:
```suggestion
  if (utils.isPOJO(path)) {
```

you are currently missing a ending `)`, so the tests fail

**Review by hasezoey — CHANGES_REQUESTED** [review]
if possible, it would also be good to have tests for this

**Review by Uzlopak — COMMENTED** [review]
fix linting issues

**Comment by Uzlopak** [comment]
Well the code is totally broken.

I assume the right way would be something like

```js
  if (utils.isPOJO(path)) {
    for (const pojoPath of Object.keys(path)) {
      this.$set(pojoPath, path[pojoPath], options)
    }
    return this
  }
```

**Review by Uzlopak — CHANGES_REQUESTED** [review]
This is not fixing the issue. I would actually prefer to fix the typings.

File: lib/document.js (hunk 1 above)
Comment (randy5235) [reply lib/document.js]: Thanks, thats what I get for using the online editor.

**Comment by randy5235** [comment]
I missed a parenthesis, Sorry I created this via the github UI. I'll get a proper update in shortly.  Regarding the fix, the function already handles this case (as stated per the documentation), it just doesn't switch the parameters as needed to apply it. If you look at the existing code below mine, you see how it handles the types, My code checks if the path parameter is an object,  if so, it moves the rest of the parameters forward essentially since val is no longer needed (as the values should be contained within the object of key/value pairs.  The existing code the currently handles the types. The issue I see here is that if passing path as an object, it will apply any given type to all keys(paths). @Uzlopak what are your thoughts on allowing a Type object of key:value pairs (Path: Type) to assign per path?

File: lib/document.js [hunk 2]
```diff
@@ -993,6 +993,12 @@ Document.prototype.overwrite = function overwrite(obj) {
  */
 
 Document.prototype.$set = function $set(path, val, type, options) {
+  if (utils.isPOJO(path)) {
```
Comment (vkarpov15) [inline lib/document.js:996]: Make this check `if (utils.isPOJO(path) && arguments.length === 2)`

**Review by vkarpov15 — CHANGES_REQUESTED** [review]
The following syntax currently works:

```
myUser.set( { path1 : 'Value', path2: 'Value'}, null, {strict: false});
```

I can see the benefit of supporting `myUser.set(obj, options)`, but I'd rather not break the current syntax.

**Pull request closed without merging**